frontend/src/metabase/lib/urls.js

```diff
@@ -27,6 +27,10 @@ export function dashboard(dashboardId) {
     return `/dashboard/${dashboardId}`;
 }
 
+export function dashboardWithAddCard(dashboardId, cardId) {
```

Comment:
Instead of this I'd suggest the same thing we do for `Urls.question()`, which takes optional `hash` and `query` arguments. e.x. `Urls.dashboard(dashboardId, { add: cardId })`.